app.js

```diff
@@ -0,0 +1,134 @@
+require('dotenv').config();
+
+let PORT = process.env.PORT;
+let express = require('express');
+let app = express();
+let db;
+const MongoClient = require('mongodb').MongoClient;
+const MONGO_URI = process.env.MONGO_URI_MLAB;
+const bodyParser = require('body-parser');
+
+MongoClient.connect(MONGO_URI, (err, database) => {
+  if(err) {
+    throw err;
+  }
+
+  db = database;
+
+  app.listen(PORT, () => {
+    console.log(`Server is up and running on port ${PORT}`);
+  });
+});
+
+app.use(bodyParser.json());
+
+app.get('/initializeDatabase', (req, res) => {
+  const users = [
+    { "userId": 1, "firstName": "Amr", "lastName": "Desouky" },
+    { "userId": 2, "firstName": "Ernie", "lastName": "Casilla" },
+    { "userId": 3, "firstName": "Ben", "lastName": "Sammons" } ];
+  const accounts = [
+    { "type": "Checking", "userId": 1, "amount": 300.00, "currency": "USD", "timestamp": Date.now() },
+    { "type": "Saving",   "userId": 1, "amount": 200.00, "currency": "USD", "timestamp": Date.now() },
+    { "type": "Checking", "userId": 2, "amount": 300.00, "currency": "EUR", "timestamp": Date.now() },
+    { "type": "Saving",   "userId": 3, "amount": 300.00, "currency": "USD", "timestamp": Date.now() } ];
+
+  db.collection('users').remove((error, result) => { });
+  db.collection('accounts').remove((error, result) => { });
+  db.collection('users').insert(users, (error, result) => { });
+  db.collection('accounts').insert(accounts, (error, result) => { });
+
+  res.send('done');
+});
+
+app.get('/account/:userID/balance', (req, res) => {
+  const userId = parseInt(req.params.userID);
+    
+  getAccountBalance('accounts', userId)
+    .then(result => {
+      return res.send(result);
+    })
+    .catch(error => {
+      return res.send('Error getAccountBalance ' + error);
+    });
+});
+
+app.put('/account/:userID/deposit', (req, res) => {
+  const userId = parseInt(req.params.userID);
+  const depositObject = req.body;
```

Comment:
use object destructuring on the incoming parameters off the body object 

`{type, amount} = req.body`

guard checks against incoming userId

Reply:
I just picked up javascript about 2 weeks ago, I didn't even know object destructing was a thing until now. I just looked it up, so thanks for helping me learn! I look forward to finding other places to use this